### ISSUE TITLE
Input loop on today build (b87d355)

### ISSUE THREAD

**Issue by marcinkoziej** [onivim/oni2#3084]
Platform: Ubuntu Linux 20.04

At one point editor gets stuck in a kind of input loop, and I can type for instance Ctrl-P but the menu will appear and disappear fast, and becomes unusable. I don't remember what I have typed to trigger this - but this was movement, editing, maybe /-searching of TypeScript file.
In the terminal log, i can see a messages being printed in a loop even if i do not do anything. I had to Ctrl-C stop the editor.
I have copied the looping part of the log and some lines before it started, hope this helps, it's here:
https://gist.github.com/marcinkoziej/1b354494db301149b019228e6dc1464e

**Comment by BenoitPingris** [comment]
I was about to open a new issue then saw yours, I have the same issue with today's build, after I `dd` in my case I have the weird behaviour with the loop.

Here are my logs if it can help debug:
[oni.log](https://github.com/onivim/oni2/files/5921220/oni.log)

**Comment by bryphe** [comment]
Thanks @marcinkoziej and @BenoitPingris for logging the issue and the details!

I'm investigating now - I can reproduce by using the `jk` -> `esc` key binding:
```
   { "key": "jk", "command": "vim.esc", "when": "insertMode" }
```

And I get caught in the same input loop as in @BenoitPingris logs. I suspect it is the same issue in the first set of logs.

Do you happen to have any custom keybindings, @marcinkoziej ? Or mappings set via `experimental.viml`?

I've narrowed it down to this PR: https://github.com/onivim/oni2/pull/3055 - looking to see the best fix

**Comment by bryphe** [comment]
Reverted #3055 in https://github.com/onivim/oni2/commit/a646cb9620abda25991de6727003a68aea6fa9e8 - kicked off a new nightly build.

Going to work on bringing it back w/ a fix & a test.

**Comment by marcinkoziej** [comment]
@bryphe just one key binding, and no experimental onesx:
```
[ { "key": "<C-W>k", "command": "view.closeEditor", "when": "editorTextFocus"}]
```
I did not customize preferences much except for:
`  "vim.useSystemClipboard": true`, tabSize, indentSize

**Comment by bryphe** [comment]
Thanks for all the details! I believe this should be fixed now in #3088 - let me know if you still see it :+1:
